fastlane/actions/get_latest_github_release_within_same_major.rb

```diff
@@ -0,0 +1,100 @@
+module Fastlane
+  module Actions
+
+    class GetLatestGithubReleaseWithinSameMajorAction < Action
+      def self.run(params)
+        repo_name = params[:repo_name]
+        UI.message("Getting latest release for #{repo_name} on GitHub within same major as the current version")
+
+        GithubApiAction.run(
+          server_url: "https://api.github.com",
+          api_token: ENV["GITHUB_API_TOKEN"],
+          api_bearer: nil,
+          http_method: 'GET',
+          path: "repos/RevenueCat/#{repo_name}/releases",
+          error_handlers: {
+            404 => proc do |result|
+              UI.error("Repository #{repo_name} cannot be found, please double check its name and that you provided a valid API token (if it's a private repository).")
+            end,
+            401 => proc do |result|
+              UI.error("You are not authorized to access #{repo_name}, please make sure you provided a valid API token.")
+            end,
+            '*' => proc do |result|
+              UI.error("GitHub responded with #{result[:status]}:#{result[:body]}")
+            end
+          }
+        ) do |result|
+          current_version = Gem::Version.new(params[:current_version])
+          json = result[:json]
+          if json.count == 0
+            UI.user_error!("Couldn't find latest GitHub release for #{params[:repo_name]}")
+            return
+          end
+          highest_version = current_version
+          json.reject {  |item| item.key?("prerelease") }.each do |item|
+            item_version = Gem::Version.new(item['tag_name'])
+            # Skip if it's not in the same major range. Ex. 3.0.0 vs 4.2.0 
+            next if item_version.canonical_segments[0] != current_version.canonical_segments[0]
```

Comment:
you could also extract the line into a function for readability, i.e.: 
```ruby
next if !same_major_range?(item_version, current_version)

...

def same_major_range?(version_a, version_b)
    version_a.canonical_segments[0] == version_b.canonical_segments[0]
end

```

Reply:
furthermore, since this is still filtering items on the list, you could add it to the filters, i.e.: 
```ruby

def item_version(item)
  Gem::Version.new(item['tag_name'])
end

highest_version = json
  .reject { |item| item.key?("prerelease") }
  .map { |item| item_version(item) }
  .select { |item_version| same_major_range?(item_version, current_version) }
  .max
```

Reply:
disclaimer: I didn't test any of that code 😅 so apologies if it doesn't work as is, but you get the idea